src/cli/serve/serve.js

```diff
@@ -209,31 +222,32 @@ export default function (program) {
     }
 
     const unknownOptions = this.getUnknownOptions();
-    await bootstrap({
-      configs: [].concat(opts.config || []),
-      cliArgs: {
-        dev: !!opts.dev,
-        envName: unknownOptions.env ? unknownOptions.env.name : undefined,
-        // no longer supported
-        quiet: !!opts.quiet,
-        silent: !!opts.silent,
-        watch: !!opts.watch,
-        runExamples: !!opts.runExamples,
-        // We want to run without base path when the `--run-examples` flag is given so that we can use local
-        // links in other documentation sources, like "View this tutorial [here](http://localhost:5601/app/tutorial/xyz)".
-        // We can tell users they only have to run with `yarn start --run-examples` to get those
-        // local links to work.  Similar to what we do for "View in Console" links in our
-        // elastic.co links.
-        basePath: opts.runExamples ? false : !!opts.basePath,
-        optimize: !!opts.optimize,
-        disableOptimizer: !opts.optimizer,
-        oss: !!opts.oss,
-        cache: !!opts.cache,
-        dist: !!opts.dist,
-      },
-      features: {
-        isCliDevModeSupported: DEV_MODE_SUPPORTED,
-      },
+    const configs = [].concat(opts.config || []);
+    const cliArgs = {
+      dev: !!opts.dev,
+      envName: unknownOptions.env ? unknownOptions.env.name : undefined,
+      // no longer supported
+      quiet: !!opts.quiet,
+      silent: !!opts.silent,
+      watch: !!opts.watch,
+      runExamples: !!opts.runExamples,
+      // We want to run without base path when the `--run-examples` flag is given so that we can use local
+      // links in other documentation sources, like "View this tutorial [here](http://localhost:5601/app/tutorial/xyz)".
+      // We can tell users they only have to run with `yarn start --run-examples` to get those
+      // local links to work.  Similar to what we do for "View in Console" links in our
+      // elastic.co links.
+      basePath: opts.runExamples ? false : !!opts.basePath,
+      optimize: !!opts.optimize,
+      disableOptimizer: !opts.optimizer,
+      oss: !!opts.oss,
+      cache: !!opts.cache,
+      dist: !!opts.dist,
+    };
+    const bootstrapScript = getBootstrapScript(cliArgs.dev);
```

Comment:
Would you mind adding a description of this diagram to help the next reader? 
![](https://user-images.githubusercontent.com/1532934/111953545-12a7d700-8ae7-11eb-9bc5-17d16438674e.png)